app/models/manageiq/providers/openstack/network_manager.rb

```diff
@@ -111,4 +112,25 @@ def create_network_router_queue(userid, options = {})
     }
     MiqTask.generic_action_with_callback(task_opts, queue_opts)
   end
+
+  def create_floating_ip(options)
+    FloatingIp.raw_create_floating_ip(self, options)
+  end
+
+  def create_floating_ip_queue(userid, options = {})
```

Comment:
How about getting a refactoring PR in first and then use it in this PR.
I agree with @Ladas that we should refrain from code getting in and hope it'll be refactored later.